### PR TITLE
Adding AWS Multiple Workspace Example

### PR DESCRIPTION
**Pull request by JDBraun**
This example provides a functioning AWS multiple workspace example. 

Included in this example are:
- **Logging:** A logging pipeline that creates all of the underlying cloud resources and Databricks resources to receive billing and audit logs.
- **Unity Catalog:** An isolated pipeline that creates the needed S3 bucket and IAM role for the root metastore. This isolates the metastore using the account level provider and decouples it from any workspaces.
- **Common Account Modules:** In this folder are commonly used account related modules that are called from each of the environments. These include: cloud provider assets (e.g. cross-account role, DBFS, and basic networking set-up) as well as environment specific Databricks assets (e.g. workspace creation, metastore assignment, and identity assignment).
- **Common Workspace Modules:** In this folder are commonly used workspace related modules that are called from each of the environments. These include cluster policies and workspace configurations as examples.
- **Databricks environment:** Three environments are included as examples that invoke the previous mentioned modules. These can stay as is, be reduced, or changed to reflect the logical set-up of the end user.

In addition:
**Reference Images**: These reference images include real screenshots as well as logical architecture diagrams of the set-up
**tfvars Examples**: To expedite adoption of this example, we've provided tfvars examples that can be copied and applied into the repo

Further improvements:
- **Metastore and identities assignment**: We can further improve this repo by building out the metastore and identities assignment into one. Then include a way to iterate over a list of account level identities instead of a single user.

### CODE REVIEW

File: examples/aws-multiple-workspaces/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    - Cloud Provider: Subsection for cloud related assets from modules and environment specifics (e.g. network peering)
+    - Databricks Account: Subsection for account related assets from modules and environment specifics (e.g. identitiy assignment)
+    - Databricks Workspace: Subsection for workspace related assets from modules and enviornment specifics (e.g. repos, notebooks, etc.)
+
+### How to use:
+- Create a .tfvars file based on the examples found in the tvfvars_examples page in each seperate pipeline (Unity Catalog, Logging, and the various Databricks environments)
+- **Recommended**: Set environment variables for your AWS and Databricks credentials
+- Terraform init (each pipeline)
+- Terraform plan (each pipeline)
+- Terraform apply (each pipeline)
```
Comment (alexott) [inline examples/aws-multiple-workspaces/README.md:34]: use verbatim: `terraform init`, ...

File: examples/aws-multiple-workspaces/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+**Note**: Please raise a git issues with any problems or concerns about the repo.
+
+### FAQ:
+- **"Error: cannot create mws credentials: MALFORMED_REQUEST: Failed credential validation checks: please use a valid cross account IAM role with permissions setup correctly. What do I do to fix this?"**
+    - This occurs after the networking configured is finalized. This is due to a race condition between the IAM role and the logging of it to the Databricks endpoint. Please re-plan and apply and it will go through. It can be mitigated with a sleep condition. We are looking at adding this sleep condition into the repo.
```
Comment (alexott) [inline examples/aws-multiple-workspaces/README.md:40]: Let's create a follow up issue to track it.

File: examples/aws-multiple-workspaces/repo/common_infrastructure/logging/logging.tf
Comment (alexott) [inline examples/aws-multiple-workspaces/repo/common_infrastructure/logging/logging.tf:1]: why do we have a `repo` folder?

**Review by alexott — CHANGES_REQUESTED** [review]
Let's structure it differently - modules will go to the top-level `modules` folder, and then in `examples` we'll just show how to call these modules.

File: examples/aws-multiple-workspaces/README.md (hunk 2 above)
Comment (JDBraun) [reply examples/aws-multiple-workspaces/README.md:40]: same as this one: https://github.com/databricks/terraform-provider-databricks/issues/1424

File: examples/aws-multiple-workspaces/repo/common_infrastructure/logging/logging.tf
Comment (JDBraun) [reply examples/aws-multiple-workspaces/repo/common_infrastructure/logging/logging.tf:1]: as the main level folder? just the initial naming convention, I'll update it

**Comment by JDBraun** [comment]
> Let's structure it differently - modules will go to the top-level `modules` folder, and then in `examples` we'll just show how to call these modules.

@alexott if I'm reading this correctly:

- reference_images
- modules (infrastructure, account, workspace)
- examples (dev, prod, qa)
- README.md

is that how you were imaging it?

**Comment by alexott** [comment]
@JDBraun I meant to have correct structure inside the repository, like this:

```
examples/
  aws-multi-workspace
  aws-multi-workspace/README.md
  aws-multi-workspace/*.tf files
modules/
  aws-module1
     README.md
     main.tf
     ...
  aws-module2
     README.md
     main.tf
     ...
```

images could be placed in the subdirectories of `examples` or in `modules`... 

I've tried to talk about this layout in this PR: https://github.com/databricks/terraform-databricks-examples/pull/111/files

**Comment by JDBraun** [comment]
> @JDBraun I meant to have correct structure inside the repository, like this:
> 
> ```
> examples/
>   aws-multi-workspace
>      README.md
>      *.tf files
> modules/
>   aws-module1
>      README.md
>      main.tf
>      ...
>   aws-module2
>      README.md
>      main.tf
>      ...
> ```
> 
> images could be placed in the subdirectories of `examples` or in `modules`...
> 
> I've tried to talk about this layout in this PR: https://github.com/databricks/terraform-databricks-examples/pull/111/files

Got it, thanks! I'll go ahead and take another stab at restructuring

**Pull request closed without merging**